features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/thresholding/FlowThresholding.java

```diff
@@ -133,78 +123,144 @@ public void run() {
         }, stepSizeMs, stepSizeMs);
     }
 
+    public long getIdleTimeoutMs() {
+        return this.idleTimeoutMs;
+    }
+
+    public void setIdleTimeoutMs(final long idleTimeoutMs) {
```

Comment:
Can we enforce an invariant that `idleTimeoutMs` must be greater than `stepSizeMs`? If `IdleTimeoutMs` is too small sessions get removed before their values are reported.

Reply:
If idle timeout is lower than step size, this would result in invalidating after doing the threshold. Which would result in strange behavior. I think that this is more a documentation issue as I can not control the order in which the setters are applied during config changes. Therefore we must allow the broken combination as temporary during config changes.

Reply:
The timer could be started in an init method that is called after the bean is completely configured. This would be the time when `idleTimoutMs` and `stepSize` is checked. There could also be a destroy method that stops the timer. This would make the bean a better OSGI inhabitant.

Reply:
It is expected to be user-configurable during runtime. So an `init` will not help here.

Reply:
Isn't that covered by OSGI? The bean gets reloaded when its configuration is changed (`update-strategy="reload"`)? I am not sure about that, though.

Reply:
I still don't see why this is a problem at all? Everything is processed at least once, right?

Reply:
How shall the configuration change happen? If the configuration file is changed on disk then blueprint will reload the container. A new `ThreadThresholding` bean gets instantiated. The timer task of the previous `ThreadThresholding` bean will run for ever.

Reply:
The timer task was not even changed in the PR... An this is about the idle timeout, not the interval of the timer.
In addition, the timer is canceled in the `close` method.

Reply:
Where is the `close` method called? The `ElasticFlowRepository` that references the `FlowThresholding` does not. (It would also make no sense, because it does not "open/create" it either.) I can create an issue for the timer task lifecycle if you prefer.

Reply:
https://github.com/OpenNMS/opennms/pull/3935/files#diff-79652c14483859c21c60d394b21ddd270fb00d23b810c3aba23d3dacbff95f08R216

Reply:
Again, this is even the wrong thread...